scripts/lmql-serve-docker.py

```diff
@@ -37,20 +39,20 @@ def build_docker_image():
 if not has_docker_image() or args.rebuild:
     build_docker_image()
 
-PORT=2223
-GPUS=all
-
-cmd = """docker run \\
-    -p $PORT:8899 \\
-    -e TRANSFORMERS_CACHE=/transformers $EXTRAS \\
-    -it --gpus $GPUS \\
-    -v $CACHE:/transformers \\
-    lmql-serve $@
- """.replace("$GPUS", args.gpus) \
-    .replace("$PORT", str(args.port)) \
-    .replace("$@", " ".join(_)) \
-    .replace("$CACHE", args.transformers_cache) \
-    .replace("$EXTRAS", f"-e EXTRA_PIP_PACKAGES='{args.extras}'" if args.extras != "" else "")
-
-print(">", cmd)
-os.system(cmd)
\ No newline at end of file
+script = r"""
+exec docker run \
+    -p "$port:8899" \
+    -e TRANSFORMERS_CACHE=/transformers ${extras:+ -e EXTRA_PIP_PACKAGES="$extras"} \
+    -it --gpus "$gpus" \
+    -v "$cache:/transformers" \
+    lmql-serve "$@"
+"""
```

Comment:
Note that we're letting the shell itself do all the substitutions instead of doing them ourselves. That's an important distinction, because when a shell substitutes a value for an argument, almost all parsing steps are skipped (the "almost" items that still take place are word splitting and glob expansion, and those are skipped if-and-only-if the expansion is in a double-quoted string, which is consistently the case here).